darts/models/torch_forecasting_model.py

```diff
@@ -200,6 +205,131 @@ def __init__(self,
         self.lr_scheduler_cls = lr_scheduler_cls
         self.lr_scheduler_kwargs = dict() if lr_scheduler_kwargs is None else lr_scheduler_kwargs
 
+        checkpoints_folder = _get_checkpoint_folder(self.work_dir, self.model_name)
+        if os.path.exists(checkpoints_folder) and len(glob(os.path.join(checkpoints_folder, "checkpoint_*"))) > 0:
+            self.load()
+
+    def load(self, best=False):
+        ''' Loads the model from a checkpoint by given name and work directory.
+
+            Parameters
+            ----------
+            best
+                Load the best performing checkpoint checkpoint instead of the latest checkpoint (the default)
+
+        '''
+        params = {k:v for k,v in self.get_params() if k not in self.immutable_args}
+        model = self.load_from_checkpoint(self.model_name, self.work_dir, best)
+        self.__dict__.update(model.__dict__)
+        self.set_params(**params)
+
+    @classmethod
+    def _get_param_names(cls):
+        """Get parameter names for the estimator"""
+        # fetch the constructor or the original constructor before
+        # deprecation wrapping if any
+        # source: scikit-learn's base.py
+        init = getattr(cls.__init__, 'deprecated_original', cls.__init__)
+        if init is object.__init__:
+            # No explicit constructor to introspect
+            return []
+
+        # introspect the constructor arguments to find the model parameters
+        # to represent
+        init_signature = inspect.signature(init)
+        # Consider the constructor parameters excluding 'self'
+        parameters = [p for p in init_signature.parameters.values()
+                      if p.name != 'self' and p.kind != p.VAR_KEYWORD]
+        for p in parameters:
+            if p.kind == p.VAR_POSITIONAL:
+                raise RuntimeError("scikit-learn estimators should always "
+                                   "specify their parameters in the signature"
+                                   " of their __init__ (no varargs)."
+                                   " %s with constructor %s doesn't "
+                                   " follow this convention."
+                                   % (cls, init_signature))
+        # Extract and sort argument names excluding 'self'
+        return sorted([p.name for p in parameters])
+
+    def get_params(self, deep=True):
+        """
+        Get parameters for this estimator.
+        source: scikit-learn's base.py
+        Parameters
+        ----------
+        deep : bool, default=Tru
```

Comment:
We might have to adapt the docstring to match our own docstrings here and for the other sklearn-based methods.